source/scripting_v3/GTA/ThrowHelper.cs

```diff
@@ -11,165 +11,223 @@
-        /// <exception cref="ArgumentOutOfRangeException">
-        /// <paramref name="value"/> is not between <paramref name="minInclusive"/> and <paramref name="maxInclusive"/>.
-        /// </exception>
-        internal static void CheckArgumentRange(string paramName, int value, int minInclusive, int maxInclusive)
+
+        /// <inheritdoc cref="CheckArgumentRange(string, int, int, int)"/>
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        internal static void CheckArgumentRange(string paramName, float value, float minInclusive, float maxInclusive)
         {
             if (value < minInclusive || value > maxInclusive)
             {
                 ThrowArgumentOutOfRangeException(paramName, value, minInclusive, maxInclusive);
             }
         }
 
-        internal static void ThrowArgumentException_Arg_CannotBeNaN(string paramName)
+        /// <inheritdoc cref="CheckArgumentRange(string, int, int, int)"/>
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        internal static void CheckArgumentRange(string paramName, double value, double minInclusive, double maxInclusive)
         {
-            throw new ArgumentException("GameClockDuration does not accept floating point Not-a-Number values.", nameof(paramName));
+            if (value < minInclusive || value > maxInclusive)
+            {
+                ThrowArgumentOutOfRangeException(paramName, value, minInclusive, maxInclusive);
+            }
         }
 
+        /// <inheritdoc cref="CheckArgumentRange(string, int, int, int)"/>
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        internal static void CheckArgumentRange(string paramName, long value, long minInclusive, long maxInclusive)
+        {
+            if (value < minInclusive || value > maxInclusive)
+            {
+                ThrowArgumentOutOfRangeException(paramName, value, minInclusive, maxInclusive);
+            }
+        }
+
+        #endregion
+
         /// <summary>
-        /// Throws an <see cref="ArgumentOutOfRangeException"/> with a parameter name for an enum value out of legal
-        /// range, which means the value is not defined.
+        /// Throws an <see cref="ArgumentException"/> for the specified parameter when a floating-point value is NaN.
         /// </summary>
-        /// <exception cref="ArgumentOutOfRangeException"/>
-        internal static void ArgumentOutOfRangeException_Enum_Value(string paramName)
+        /// <param name="paramName">The name of the parameter.</param>
+        /// <exception cref="ArgumentException"/>
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        internal static void ThrowArgumentExceptionForNaN(string paramName)
         {
-            throw new ArgumentOutOfRangeException(paramName, "Enum value was out of legal range.");
+            throw new ArgumentException("Floating-Point value is not allowed to be NaN!", paramName);
```

Comment:
A message that tells you a passed floating-point parameter isn't allowed to be NaN would be better than the current one, which might make some users think no floating-point values can be NaN.